### ISSUE TITLE
`unix_timestamp(date)` doesn't respect a session time zone

### ISSUE THREAD

**Issue by towny53** [dolthub/dolt#6712]
According to the MySQL documentation:

> If [UNIX_TIMESTAMP()](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_unix-timestamp) is called with a date argument, it returns the value of the argument as seconds since '1970-01-01 00:00:00' UTC. **The server interprets date as a value in the session time zone and converts it to an internal Unix timestamp value in UTC.**

---

These are the results I get on a plain MySQL 8 database:

``` 
SET time_zone = '+07:00';
SELECT UNIX_TIMESTAMP('2023-09-25 07:02:57');
```

| UNIX\_TIMESTAMP('2023-09-25 07:02:57') |
|--------------------------------------:|
| 1695600177 |

``` 
SET time_zone = '+07:00';
SELECT UNIX_TIMESTAMP(CONVERT_TZ('2023-09-25 07:02:57', '+00:00', @@session.time_zone));
```

| UNIX\_TIMESTAMP(CONVERT\_TZ('2023-09-25 07:02:57', '+00:00', @@session.time\_zone)) |
|---------------------------------------------------------------------------------:|
| 1695625377 |

``` 
SET time_zone = '+00:00';
SELECT UNIX_TIMESTAMP('2023-09-25 07:02:57');
```

| UNIX\_TIMESTAMP('2023-09-25 07:02:57') |
|--------------------------------------:|
| 1695625377 |

[fiddle](https://dbfiddle.uk/VYDowrQQ)

---

I get different results when I run the same queries on a Dolt database:

``` 
SET time_zone = '+07:00';
SELECT UNIX_TIMESTAMP('2023-09-25 07:02:57');
```

| UNIX\_TIMESTAMP('2023-09-25 07:02:57') |
|--------------------------------------:|
| 1695625377 |

``` 
SET time_zone = '+07:00';
SELECT UNIX_TIMESTAMP(CONVERT_TZ('2023-09-25 07:02:57', '+00:00', @@session.time_zone));
```

| UNIX\_TIMESTAMP(CONVERT\_TZ('2023-09-25 07:02:57', '+00:00', @@session.time\_zone)) |
|---------------------------------------------------------------------------------:|
| 1695650577 |

``` 
SET time_zone = '+00:00';
SELECT UNIX_TIMESTAMP('2023-09-25 07:02:57');
```

| UNIX\_TIMESTAMP('2023-09-25 07:02:57') |
|--------------------------------------:|
| 1695625377 |

**Comment by timsehn** [comment]
This is known: #3119. I'm not sure which one to keep open. I'll let @fulghum decide. Likely this one since it's not from one of us.

**Comment by timsehn** [comment]
Also, moved the issue to the Dolt repository for visibility.

**Comment by ydynnikov** [comment]
@timsehn this one is from us :)

**Comment by fulghum** [comment]
Hi @towny53, thank you for taking the time to report this issue! 

Issue https://github.com/dolthub/dolt/issues/3119 is related, but has a larger scope. I think we may be able to get this specific case with `UNIX_TIMESTAMP` working correctly for you with a narrow change to the `UNIX_TIMESTAMP` function, whereas issue https://github.com/dolthub/dolt/issues/3119 requires changes in many places to properly handle MySQL rules for converting `TIMESTAMP` fields when they go in and out of storage. 

So, for now, we'll leave both issues open and track the more targeted fix here so that we can unblock you as fast as possible. @jennifersp is digging into the fix for the `UNIX_TIMESTAMP` function, and we should know more by this afternoon if there are other snags with `TIMESTAMP` handling that we run into.

**Comment by fulghum** [comment]
@towny53 – thank you for reporting this issue so we could fix it! 🙏  The fix has now been released in [Dolt version 1.16.5](https://github.com/dolthub/dolt/releases/tag/v1.16.5)

Please let us know if you hit any other problems with Dolt and we'll be happy to help!
